### PR TITLE
RExec.connect: handle protocol version negotiation

### PR DESCRIPTION
**Pull request by reynir**
Continuation of #56.

Adds debug message the other place where protocol version is negotiated. In the case of version > 2 it would work without this patch. It fails if the other end wants to use version 1 for good measure.

### CODE REVIEW

**Comment by hannesm** [comment]
The "fail if other ends wants version 1" looks fine to me; but I'm not a big fan of the conditionally print a debug message -- and would either use a warning there, or keep the old code (an info message printing the remote version).

**Comment by talex5** [comment]
It's not really a warning - there's nothing wrong with the other end offering a newer version.

How about an `info` message that always shows the version offered and the version chosen?

**Comment by reynir** [comment]
The old code was wrong in its log message, though, in the case of the other end wanting to use protocol version 3. Then it would log us using protocol version 3 even though we send protocol version 2 thereby forcing the protocol version to 2.

**Comment by reynir** [comment]
The new commit unconditionally logs the protocol version negotiation in both places.

**Comment by hannesm** [comment]
that looks good - though I'd condense the two `Log` in `connect` into a single on the Info level -- and also use the Info level in `with_flow`

**Comment by reynir** [comment]
Yes, I'll fix it up. The log message still says "newer" even though that's not necessarily true.

**Review by hannesm — APPROVED** [review]

File: lib/rExec.ml [hunk 1]
```diff
@@ -217,9 +217,9 @@ let with_flow ~ty ~domid ~port fn =
           recv_hello client >>= function
           | version when version < 2l -> fail (error "Unsupported qrexec version %ld" version)
           | version ->
-            if version > 2l
-            then Log.debug (fun f -> f "Other end wants to use newer protocol %lu, \
-                                        continuing with version 2" version);
+            Log.info (fun f -> f "client connected,\
```
Comment (talex5) [inline lib/rExec.ml:220]:
```suggestion
            Log.info (fun f -> f "client connected, \
```

Comment (reynir) [reply lib/rExec.ml:220]: Thanks!

**Pull request merged**